PyNite/Member3D.py

```diff
@@ -1764,4 +1764,83 @@ def _segment_member(self, combo_name='Combo 1'):
 
                                 SegmentsY[i].V1 += (f1[2] + f2[2])/2*(x2 - x1)
                                 SegmentsY[i].M1 += (x1 - x2)*(2*f1[2]*x1 - 3*f1[2]*x + f1[2]*x2 + f2[2]*x1 - 3*f2[2]*x + 2*f2[2]*x2)/6
-                            
\ No newline at end of file
+
+    @property
+    def nodes(self):
+        return (self.i_node, self.j_node)
+
+    @property
+    def coordinates(self):
+        return (node.coordinates for node in self.nodes)
+
+    def intersection(self, other, tolerance=1e-3, virtual=False):
+        if virtual:
+            return self.intersection_virtual(other, tolerance)
+
+        return self.intersection_real(other, tolerance)
+
+    def intersection_virtual(self, other, tolerance=1e-3):
```

Comment:
I'm trying to comment this code to bring it into the main/master branch of PyNite. I'm not sure what this method does. The math in here is pretty abstract to me. Would you mind providing some comments to help guide me through what's taking place?

Reply:
Yes I will be back shortly with an explanation.

Reply:
This is a port of the `intersection_line_line_3d` function from [mozman's ezdxf](https://github.com/mozman/ezdxf/blob/3a4be02d937dc5744785763dd7a7d2a8699eef19/src/ezdxf/math/_construct.py#L146). There is a reference to the math involved [here](https://www.realtimerendering.com/intersections.html#I304).

The math checks for intersections in two rays (virtual intersections). If the rays do not intersect, we can assume that the line segments that make up those rays will not intersect either. If there is a virtual intersection, then we determine if there is a pair of the line segments that exist at the point of the virtual intersection. If not, the line segments have the potential to intersect, but ultimately do not. If there is, we have found a real intersection.

Reply:
Virtual intersections denote intersections that occur between two rays.
Real intersections denote intersection that occur between two line segments.